File: day 1/second_largest.py
class Solution:
    def getSecondLargest(self, arr):
        n = len(arr)
        maxi = arr[0]
        smax = -1

        for i in range(1, n):
            if arr[i] > maxi:
                smax = maxi
                maxi = arr[i]
            elif arr[i] > smax and arr[i] < maxi:
                smax = arr[i]
        return smax


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        ob = Solution()
        ans = ob.getSecondLargest(arr)
        print(ans)

File: day 1/second_largest.cpp
#include <bits/stdc++.h>

using namespace std;

// User function template for C++
class Solution
{
public:
	// Function returns the second
	// largest elements
	int getSecondLargest(vector<int> &arr)
	{
		int n = arr.size();
		int max = arr[0], smax = -1;

		for (int i = 1; i < n; i++)
		{
			if (arr[i] > max)
			{
				smax = max;
				max = arr[i];
			}
			else if (arr[i] > smax and arr[i] < max)
			{
				smax = arr[i];
			}
		}

		return smax;
	}
};

int main()
{
	int t;
	cin >> t;
	cin.ignore();
	while (t--)
	{
		vector<int> arr;
		string input;
		getline(cin, input);
		stringstream ss(input);
		int number;
		while (ss >> number)
		{
			arr.push_back(number);
		}
		Solution ob;
		int ans = ob.getSecondLargest(arr);
		cout << ans << endl;
		cout << "~" << endl;
	}
	return 0;
}
